src/power_grid_model/core/power_grid_dataset.py

```diff
@@ -466,22 +466,15 @@ def _filter_attributes(self, attributes):
         for key in keys_to_remove:
             del attributes[key]
 
-    def _filter_with_option(self):
-        if self._data_filter is ComponentAttributeFilterOptions.RELEVANT:
-            for attributes in self._data.values():
-                self._filter_attributes(attributes)
-
```

Comment:
This is removed due to the fact that `self._data_filter` is a processed version that is always a aggregated version of input `data_filter`.